plugins/woocommerce/src/StoreApi/Utilities/QuantityLimits.php

```diff
@@ -92,51 +126,32 @@ public function limit_to_multiple( int $number, int $multiple_of, string $roundi
 	/**
 	 * Check that a given quantity is valid according to any limits in place.
 	 *
-	 * @param integer           $quantity Quantity to validate.
-	 * @param \WC_Product|array $cart_item Cart item.
+	 * @param integer $quantity Quantity to validate.
+	 * @param array   $cart_item Cart item.
 	 * @return \WP_Error|true
 	 */
 	public function validate_cart_item_quantity( $quantity, $cart_item ) {
-		$limits = $this->get_cart_item_quantity_limits( $cart_item );
+		$limits  = $this->get_cart_item_quantity_limits( $cart_item );
+		$product = $cart_item['data'];
 
 		if ( ! $limits['editable'] ) {
-			return new \WP_Error(
-				'readonly_quantity',
-				__( 'This item is already in the cart and its quantity cannot be edited', 'woocommerce' )
-			);
+			/* translators: 1: product name */
```

Comment:
When are these errors thrown? During set quantity?

Reply:
Updating qty via the API yeah.